server/src/main/java/org/opensearch/node/Node.java

```diff
@@ -789,14 +789,19 @@ protected Node(
                 )
                 .collect(Collectors.toList());
 
+            final List<ActionPlugin> scopeProtectedActionPlugin = pluginsService.filterPlugins(ActionPlugin.class)
```

Comment:
I like this wrapping technique.  It doesn't require adding scope checks all over the codebase, and puts them inside of a easier to code review place.

However, it also means that these checks can be missed.  Might warrant a way to check that this enforcement is properly applied everywhere.